core/src/main/java/org/radargun/stages/cache/background/PrivateLogLogic.java

```diff
@@ -31,6 +33,7 @@
    // Keys modified during current transaction, should be recorded to timestamps when
    // the transaction is committed
    private final Collection<KeyOperationPair> txModifications = new ArrayList<>(Math.max(0, transactionSize));
+   private final Map<Long, Long> maxPrunedOperationIds = new HashMap<>(2);
```

Comment:
Fixed, restricted only to transactional scenarios (for non-tx it doesn't make sense as timestamps are updated after each write/remove).